docs/tools/testing/icetea.md

```diff
@@ -0,0 +1,188 @@
+## Icetea test framework
+
+Icetea is an automated testing framework for Mbed development.
+It automates the process of flashing Mbed boards, running tests
+and accumulating test results into reports.
+Developers use it for local development as well as for
+automation in a Continuous Integration Environment.
+
+When testing [`Mbed OS`](https://www.mbed.com/en/platform/mbed-os/)
+Icetea allows you to execute commands remotely via
+the command line interface (`CLI`) in a device under test (`DUT`).
+The interface between the test framework and `DUT` can be
+for example UART or stdio.
+
+More detailed documentation on the tool is available
+[here in rst format](https://github.com/ARMmbed/icetea/tree/master/doc-source)
+and [here in markdown format](https://github.com/ARMmbed/icetea/tree/master/doc).
+
+### Prerequisites
+Icetea supports Linux (Ubuntu preferred), Windows and OS X. Our main target is Linux.
+We support both Python 2.7 and 3.5 or later. Some OS specific prerequisites below:
+
+* Linux
+    * python-dev and python-lxml
+        `sudo apt-get install python-dev python-lxml`
+    * In order to run test cases with hardware in Linux without sudo rights:
+        ```
+        sudo usermod -a -G dialout username
+        Log out & log in back to Linux
+        ```
+        This command will add the user 'username' to the 'dialout' group and
+        grant the required permissions to the USB ports.
+* OS X
+    * [XCode developer tools](http://osxdaily.com/2014/02/12/install-command-line-tools-mac-os-x/)
+    * [MacPorts](https://www.macports.org/install.php)
+    * [lxml](http://lxml.de/installation.html#installation):
+        `STATIC_DEPS=true sudo pip install lxml`
+
+* Windows
+    * python-lxml installation is problematic on Windows since
```

Comment:
Capitalize Python unless it's the name of a module, in which case I'd put it in ``